docs-src/design-decisions.md

```diff
@@ -0,0 +1,44 @@
+# Design Decisions
+
+Throughout the design of these APIs some significant design decisions have been
+made. This provides some background on the factors considered when making these
+decisions.
+
+## GatewayClass Scope
+
+### Problem
+There were a number of compelling reasons to make GatewayClass a
+namespace-scoped resource. 
+
+### Advantages of a Namespace-Scoped GatewayClass
+* Less potential for naming conflicts among different implementations.
+* Enables using Service API on a management cluster designed to provision and
+  manage other clusters and infrastructure needed by other clusters.
+* Easier to provide simple and secure defaults (only allow Gateways and Routes
+  in the same namespace by default).
+* GatewayClass parameters can live in the same namespace.
```

Comment:
If you consider "the root" and "each namespace" as "contexts", this statement is moot -- why can't GatewayClass parameters exist in the root context?

Reply:
I've removed this point and added a somewhat similar one:

>Namespaces could be used to separate different GatewayClasses, enabling admins
  to more closely restrict who can modify individual GatewayClasses.

Assuming GatewayClass parametersRef is still a local object reference, this would allow admins to easily restrict who could modify certain GatewayClasses _and_ their parameters. With a cluster-scoped class resource, users can either modify all GatewayClasses + parameters or none.